src/prefix-color-selector.spec.ts

```diff
@@ -0,0 +1,172 @@
+                'blue',
+                'green',
+                'blue',
+                'green',
+                'blue',
+            ],
+        },
+        'considers the Bright variants of colors equal to the normal colors to avoid similar colors':
+            {
+                acceptableConsoleColors: ['greenBright', 'blueBright', 'green', 'blue', 'magenta'],
+                customColors: ['green', 'blue', 'auto'],
+                expectedColors: [
+                    // Custom colors
+                    'green',
+                    'blue',
+
+                    // Picks auto colors, not repeating green and blue colors and variants initially
+                    'magenta',
+
+                    // Picks auto colors
+                    'greenBright',
+                    'blueBright',
+                    'green',
+                    'blue',
+                    'magenta',
+                ],
+            },
+    };
+    it.each(Object.entries(customTests))(
+        '%s',
+        (_, { acceptableConsoleColors, customColors, expectedColors }) => {
+            if (acceptableConsoleColors) {
+                jest.spyOn(PrefixColorSelector, 'ACCEPTABLE_CONSOLE_COLORS', 'get').mockReturnValue(
+                    acceptableConsoleColors
+                );
+            }
+            const prefixColorSelector = new PrefixColorSelector(customColors);
+            const prefixColorSelectorValues = expectedColors.map(() =>
+                prefixColorSelector.getNextColor()
+            );
+
+            expect(prefixColorSelectorValues).toEqual(expectedColors);
+        }
+    );
+
+    const autoTests = {
+        'does not repeat consecutive colors when last prefixColor is auto': false,
+        'handles when more individual auto prefixColors exist than acceptable console colors': true,
+    };
+    it.each(Object.entries(autoTests))('%s', (_, map) => {
+        // Pick auto colors over 2 sets
+        const expectedColors: string[] = [
+            ...PrefixColorSelector.ACCEPTABLE_CONSOLE_COLORS,
+            ...PrefixColorSelector.ACCEPTABLE_CONSOLE_COLORS,
+        ];
+
+        const prefixColorSelector = new PrefixColorSelector(
+            map ? expectedColors.map(() => 'auto') : ['auto']
+        );
+
+        expectedColors.reduce((previousColor, currentExpectedColor) => {
+            const actualSelectedColor = prefixColorSelector.getNextColor();
+            expect(actualSelectedColor).not.toBe(previousColor); // No consecutive colors
+            expect(actualSelectedColor).toBe(currentExpectedColor); // Expected color
+            return actualSelectedColor;
+        }, '');
+    });
+});
+
+describe('PrefixColorSelector#ACCEPTABLE_CONSOLE_COLORS', () => {
+    it('has more than 1 auto color defined', () => {
+        // ! Code assumes this always has more than one entry, so make sure
```

Comment:
make sure to...?

Reply:
Thanks! This is comment is now clarified 👍